### PR TITLE
3575 paginate organisations index

### PR DESCRIPTION
**Pull request by gpeng**
### Context

The organisations support page in publish loads all organisations and is consequently very slow and ties up the API for too long per request. We are going to add pagination to the page.

### Changes proposed in this pull request

* Add pagination to `/api/v2/organisations`
* Add ordering by name as this was previously being done in the client app but can't be now due to pagination

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by gpeng** [comment]
[[M] Paginate org#support page](https://trello.com/c/VvZx5lWw/3575-m-paginate-orgsupport-page)

File: app/controllers/api/v2/organisations_controller.rb [hunk 1]
```diff
@@ -2,16 +2,24 @@ module API
   module V2
     class OrganisationsController < API::V2::ApplicationController
       def index
+        per_page = params[:page][:per_page] || 10
```
Comment (asmega) [inline app/controllers/api/v2/organisations_controller.rb:5]: move to a private method?

File: app/controllers/api/v2/organisations_controller.rb [hunk 2]
```diff
@@ -2,16 +2,24 @@ module API
...
         if params_includes_provider?
           current_recruitment_cycle = RecruitmentCycle.current
           @organisations = @organisations.includes(:providers, :users, :nctl_organisations)
                              .where(provider: { recruitment_cycle_id: current_recruitment_cycle.id })
+
         end
 
-        render jsonapi: @organisations, include: params[:include],
-          fields: { providers: %i[provider_code provider_name], users: %i[email sign_in_user_id first_name last_name] }
+        render jsonapi: paginate(@organisations.order(:name), per_page: per_page),
```
Comment (misaka) [inline app/controllers/api/v2/organisations_controller.rb:16]: I notice that we have an N+1 query in here ...

```
2020-06-24 21:10:19.163050 D [96597:puma threadpool 002] (0.821ms) ActiveRecord -- { :sql => "SELECT COUNT(*) FROM \"course\" WHERE \"course\".\"provider_id\" = $1 AND \"course\".\"discarded_at\" IS NULL", :binds => { :provider_id => 13996 } }
2020-06-24 21:10:19.164485 D [96597:puma threadpool 002] (0.414ms) ActiveRecord -- { :sql => "SELECT COUNT(*) FROM \"course\" WHERE \"course\".\"provider_id\" = $1 AND \"course\".\"discarded_at\" IS NULL", :binds => { :provider_id => 13329 } }
2020-06-24 21:10:19.165423 D [96597:puma threadpool 002] (0.307ms) ActiveRecord -- { :sql => "SELECT COUNT(*) FROM \"course\" WHERE \"course\".\"provider_id\" = $1 AND \"course\".\"discarded_at\" IS NULL", :binds => { :provider_id => 14190 } }
...
```

one for each provider. Paging is helping out, but there's also [`Provider.include_course_count`](https://github.com/DFE-Digital/teacher-training-api/blob/master/app/models/provider.rb#L157) which should optimise this, if it'll work in this query. Is it worth adding it in while we're making changes here?

Comment (gpeng) [reply app/controllers/api/v2/organisations_controller.rb:16]: That's coming from the `meta` block in the `SerializableProvider`. I think we need to get rid of that which is probably beyond the scope of this PR. Did you have a way of hooking in `Provider.include_course_count`?

File: app/controllers/api/v2/organisations_controller.rb (hunk 1 above)
Comment (gpeng) [reply app/controllers/api/v2/organisations_controller.rb]: Good shout. I've done that, and fixed the failure it was also causing. 👍

**Pull request merged**